### ISSUE TITLE
m2eclipse complains about unknown tycho executions

### ISSUE THREAD

**Issue by reckart** [eclipse/tycho#945]
When importing a module that uses the Tycho `target-platform-configuration` plugin, a popup opens in Eclipse saying that the plugin does not declare its mappings to the m2eclipse build lifecycle.

<img width="714" alt="Screenshot 2022-05-06 at 10 49 40" src="https://user-images.githubusercontent.com/1410238/167099291-49560ce4-c64e-49f6-a33f-f8c66d0ab7ee.png">

**Comment by laeubi** [comment]
Yes this is an open issue see #146

**Comment by reckart** [comment]
Thanks. I had searched for `target-platform-configuration` to find an existing relevant issue but didn't see this one.

**Comment by laeubi** [comment]
No problem, I think we should try to address this but I'm not sure if it was decided to do this at Tycho or at m2e @mickaelistria @HannesWell what do you think?

**Comment by HannesWell** [comment]
> > I think in a first step the target-platform-configuration could just get a lifecycle mapping embedded that just disables its execution.
> 
> There are other mappings we'd like to address as well.

Yes. It should not be necessary to have the m2e-lifecycle-mapping configuration for Tycho-plugins in a pom.xml.
IMHO Tycho and M2E should work seamlessly together since it comes from the same people. :)

> 
> > The plan you mention in the linked issue to really use that configuration in the workspace seems to be a big one for me and I'm not sure if this is conceptually possible/useful/worth it.
> 
> Forget about the plan :-D We might start to provide mappings as a first step, but should also think about what might be still useful instead of simply disable anything... e.g. it would be cool if the DS plugin could actually execute, so if I enable it in my pom I don't need to enable it in PDE...

Alright, done. :D The DS-plugin is another story and for the compontent.xml generation I don't see big issues, but I wonder if it would be possible to enhance the Plug-ins classpath via a m2e-lifecycle-mapping to add the annotation jars.

**Comment by mickaelistria** [comment]
https://www.eclipse.org/m2e/documentation/m2e-making-maven-plugins-compat.html tells which files to use directly in Maven plugins to just let Eclipse ignore them.

**Comment by laeubi** [comment]
- No marketplace entries found to handle tycho-ds-plugin:3.0.0-SNAPSHOT:declarative-services in Eclipse.  Please see Help for more information.
- No marketplace entries found to handle tycho-eclipserun-plugin:3.0.0-SNAPSHOT:eclipse-run in Eclipse.  Please see Help for more information.

**Comment by HannesWell** [comment]
- No marketplace entries found to handle tycho-compiler-plugin:2.7.2:validate-classpath in Eclipse.  Please see Help for more information.

**Comment by HannesWell** [comment]
I plan to work on this issue in the next days and should have enough time on Saturday to finish it.
@laeubi is this quick enough to be considered for 2.7.4? I would really like to kick off the Tycho-Connector from some setups knowing that there is a real replacement.

**Comment by laeubi** [comment]
> is this quick enough to be considered for 2.7.4?

Why not :-) I assume this would only be a xml file per plugin.

> I would really like to kick off the Tycho-Connector from some setups knowing that there is a real replacement.

Regarding this, you can already do this, the tycho connector don't do anything useful and the latest m2e release only warns about missing connectors, so actually there is not much to gain here.

**Comment by HannesWell** [comment]
> > is this quick enough to be considered for 2.7.4?
> 
> Why not :-) I assume this would only be a xml file per plugin.

Great. I assume the same, but also would like to test this. But if nothing unforeseen happens, this should be completed tomorrow.

> 
> > I would really like to kick off the Tycho-Connector from some setups knowing that there is a real replacement.
> 
> Regarding this, you can already do this, the tycho connector don't do anything useful and the latest m2e release only warns about missing connectors, so actually there is not much to gain here.

I thought about that as well, nevertheless it would be better to have the mappings present to also fix the warnings. And getting rid of the Tycho connector increased my motivation to address this issue. :D

**Comment by laeubi** [comment]
Once this is done we should also remove that from the official catalogue as well.

**Comment by HannesWell** [comment]
> Once this is done we should also remove that from the official catalogue as well.

Agree. With 2.0 one won't be able to install them any ways.
